modules/wsl-distro.nix

```
@@ -108,19 +108,23 @@ with lib; {
 
           systemd = {
             # Disable systemd units that don't make sense on WSL
-            services = {
-              # no virtual console to switch to
-              "serial-getty@ttyS0".enable = false;
-              "serial-getty@hvc0".enable = false;
-              "getty@tty1".enable = false;
-              "autovt@".enable = false;
-              firewall.enable = false;
-              systemd-resolved.enable = false;
-              # system clock cannot be changed
-              systemd-timesyncd.enable = false;
-              # no udev devices can be attached
-              systemd-udevd.enable = false;
-            };
+            services = mkMerge [
+              {
+                # no virtual console to switch to
+                "serial-getty@ttyS0".enable = false;
+                "serial-getty@hvc0".enable = false;
+                "getty@tty1".enable = false;
+                "autovt@".enable = false;
+                firewall.enable = false;
+                # system clock cannot be changed
+                systemd-timesyncd.enable = false;
+                # no udev devices can be attached
+                systemd-udevd.enable = false;
+              }
+              (mkIf config.wsl.wslConf.network.generateResolvConf {
+                systemd-resolved.enable = false;
+              })
```

```suggestion
              systemd-resolved.enable = lib.mkDefault config.wsl.wslConf.network.generateResolvConf;
```
Wouldn't that also work?

---

If it does, we wouldn't even need the mkMerge

---

Apart from the logic error of wanting to negate it to `(!config.wsl.wslConf.network.generateResolvConf)` it does work